### PR TITLE
Add invitations INDEX action

### PR DESCRIPTION
**Pull request by enricostano**
With this PR we're adding 2 new endpoints:

`GET /api/v1/invitations?group_id=:group_id`

This endpoint will return all the invitations for a given (and authorized) group.
This endpoint requires a `group_id` query parameter.

```
POST /api/v1/invitations
{
  "group_id": 1,
  "email": "new_user@hola.com"
}
```

This endpoint will create a new invitation with the given `email` to the given (and authorized) group.
This endpoint requires `group_id` and `email` parameters in the request's body.

Furthermore we fix an existing bug in the default behavior for `show` actions: https://github.com/coopdevs/katuma/commit/86a3a62b9a1ec0cb276201abc3c13f442eff744e

### CODE REVIEW

File: engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb [hunk 1]
```diff
@@ -72,7 +87,8 @@ def accept_params
         end
 
         def load_group
-          @group = ::Onboarding::Group.find_by_id(bulk_params[:group_id])
+          group_id = bulk_params[:group_id] || groups_params[:group_id]
```
Comment (andresgutgon) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: Why this?

File: engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb [hunk 2]
```diff
@@ -2,10 +2,20 @@ module Onboarding
...
+        before_action :authenticate, only: [:index, :bulk]
+        before_action :load_group, only: [:index, :bulk]
         before_action :load_invitation, only: [:show, :accept]
 
+        # GET /api/v1/invitations?group_id=:id
+        #
+        def index
+          authorize @group
+
+          invitations = Invitation.where(group_id: @group.id)
```
Comment (andresgutgon) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: So you can get all invitations of all your groups. Right?

For me (Katuma UI) is not important. But maybe in the future (Do not change now please) is interesting as an API consumer

**Comment by andresgutgon** [comment]
:+1:

File: engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb [hunk 3]
```diff
@@ -2,10 +2,20 @@ module Onboarding
...
-        before_action :authenticate, only: [:bulk]
-        before_action :load_group, only: [:bulk]
+        before_action :authenticate, only: [:index, :bulk]
+        before_action :load_group, only: [:index, :bulk]
         before_action :load_invitation, only: [:show, :accept]
 
+        # GET /api/v1/invitations?group_id=:id
+        #
+        def index
+          authorize @group, :show?
```
Comment (enricostano) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: where is defined the `:show?` policy?

Comment (andresgutgon) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: In the group policy

Comment (enricostano) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: this is broken :sob: I need to check why this returns `true` all the time.

**Comment by andresgutgon** [comment]
@enricostano today I couldn't do send invitation endpoint. If you want to to it would be great :smile:

File: engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb (hunk 3 above)
Comment (enricostano) [inline engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb]: fixed by https://github.com/coopdevs/katuma/commit/86a3a62b9a1ec0cb276201abc3c13f442eff744e

**Comment by andresgutgon** [comment]
Already merged front end :tada:

**Pull request merged**